example_project/mappers/CreateBar.rb

```diff
@@ -509,11 +509,20 @@ def time_mapping(time)
             
             # create a bar building, will have spaces tagged with individual space types given the
             # input building types
+            geojson_file = scenario.feature_file.path
+            all_features = URBANopt::GeoJSON::GeoFile.from_file(geojson_file)
+            feature = all_features.get_feature_by_id(feature_id)
+
+            aspect_ratio = feature.calculate_aspect_ratio
+            perimeter = feature.calculate_perimeter(feature)
```

Comment:
is there a reason why you can't use the footprint_perimeter from the geojson file?

Reply:
I added methods to calculate the footprint perimeter from GeoJSON coordinates because footprint_perimeter was not a required property. I can add a statement to check if it exists in the geojson file and use that, othervise calculate from the geojson coordinates.

Reply:
I think that would be good.  footprint_perimeter is auto-calculated when a footprint is drawn in the UI, so it should exist in some cases.